chain/indexer/src/handlers.rs

```diff
@@ -650,35 +653,62 @@ where
 
         match event {
-            HoprTicketWinningProbabilityOracleEvents::TicketMinWinningProbability(update) => {
+            HoprWinningProbabilityOracleEvents::WinProbUpdatedFilter(update) => {
+                let mut encoded_old: EncodedWinProb = Default::default();
+                encoded_old.copy_from_slice(&update.old_win_prob.to_be_bytes()[1..]);
+                let old_minimum_win_prob = win_prob_to_f64(&encoded_old);
+
+                let mut encoded_new: EncodedWinProb = Default::default();
+                encoded_new.copy_from_slice(&update.new_win_prob.to_be_bytes()[1..]);
+                let new_minimum_win_prob = win_prob_to_f64(&encoded_new);
+
                 trace!(
-                    "on_ticket_minimum_win_prob_updated - old: {:?} - new: {:?}",
-                    update.0.to_string(),
-                    update.1.to_string()
+                    "on_ticket_minimum_win_prob_updated - old: {old_minimum_win_prob} - new: {new_minimum_win_prob}",
                 );
 
                 self.db
-                    .set_minimum_incoming_ticket_win_prob(Some(tx), win_prob_to_f64(update.1))
+                    .set_minimum_incoming_ticket_win_prob(Some(tx), new_minimum_win_prob)
                     .await?;
 
-                info!("minimum ticket winning probability has been set to {}", update.1);
+                info!("minimum ticket winning probability has been updated {old_minimum_win_prob} -> {new_minimum_win_prob}");
+
+                // If the old minimum was less strict, we need to mark of all the
+                // tickets below the new higher minimum as rejected
+                if old_minimum_win_prob < new_minimum_win_prob {
+                    let mut selector: Option<TicketSelector> = None;
+                    for channel in self.db.get_incoming_channels(tx.into()).await? {
+                        selector = selector
+                            .map(|s| s.also_on_channel(channel.get_id(), channel.channel_epoch))
+                            .or_else(|| Some(TicketSelector::from(channel)));
+                    }
+                    // Reject unredeemed tickets on all the channels with win prob lower than the new one
+                    if let Some(selector) = selector {
+                        let num_rejected = self
+                            .db
+                            .mark_tickets_as(
+                                selector.with_winning_probability_lt(encoded_new),
+                                TicketMarker::Rejected,
+                            )
+                            .await?;
+                        info!("{num_rejected} unredeemed tickets were rejected because the minimum winning probability has been increased");
+                    }
+                }
             }
             _ => {
-                // ignore other events
+                // Ignore other events
             }
         }
         Ok(None)
-    }*/
+    }
```

Comment:
Can you explain how this could happen?

Reply:
Your reasoning is flawed, since `also_on_channel` does not return an Option, but the TicketSelector type. Therefore, the value cannot be remapped in `map` to `None`.